packages/design-system/ui/src/components/date-picker/date-picker.tsx

```diff
@@ -219,26 +252,80 @@ const formatDate = (date: Date, includeTime?: boolean) => {
 }
 
 type CalendarProps = {
+  /**
+   * The year to start showing the dates from.
+   */
   fromYear?: number
+  /**
+   * The year to show dates to.
+   */
   toYear?: number
+  /**
+   * The month to start showing dates from.
+   */
   fromMonth?: Date
+  /**
+   * The month to show dates to.
+   */
   toMonth?: Date
+  /**
+   * The day to start showing dates from.
+   */
   fromDay?: Date
+  /**
+   * The day to show dates to.
+   */
   toDay?: Date
+  /**
+   * The date to show dates from.
+   */
   fromDate?: Date
+  /**
+   * The date to show dates to.
+   */
   toDate?: Date
 }
 
 interface PickerProps extends CalendarProps {
+  /**
+   * The class name to apply on the date picker.
+   */
   className?: string
+  /**
+   * Whether the date picker's input is disabled.
+   */
   disabled?: boolean
+  /**
+   * Whether the date picker's input is required.
+   */
   required?: boolean
+  /**
+   * The date picker's size.
+   */
   size?: "small" | "base"
+  /**
+   * Whether to show a time picker along with the date picker.
+   */
   showTimePicker?: boolean
+  /**
+   * The element ID of the date picker input.
+   */
   id?: string
+  /**
+   * Whether the selected value is invalid.
+   */
   "aria-invalid"?: boolean
+  /**
+   * The label of the date picker.
+   */
   "aria-label"?: string
+  /**
+   * The ID of the element labeling the date picker.
+   */
   "aria-labelledby"?: string
+  /**
+   * Whether the date picker is required.
+   */
   "aria-required"?: boolean
```

Comment:
Don't think these (aria-) require documentation, they are destructured to pass onto the child element (actual input)